catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -598,56 +574,53 @@ protected ResourceInfo getResourceInfo(
   }
 
   private ResourceInfo getResourceInfo(
-      QueryRequest queryRequest,
-      URI uri,
+      QueryResponse queryResponse,
+      Object queriedAttributeValue,
       Map<String, Serializable> requestProperties,
-      StringBuilder federatedSite,
-      boolean fanoutEnabled)
-      throws ResourceNotFoundException, UnsupportedQueryException, FederationException {
+      StringBuilder federatedSite)
+      throws ResourceNotFoundException, UnsupportedQueryException {
     Metacard metacard;
-    URI resourceUri = uri;
-    QueryResponse queryResponse = queryOperations.query(queryRequest, null, true, fanoutEnabled);
-    if (queryResponse.getResults().isEmpty()) {
-      throw new ResourceNotFoundException(
-          "Could not resolve source id for URI by doing a URI based query.");
+    URI resourceUri;
+
+    if (queryResponse.getResults().size() > 1) {
+      throw new UnsupportedQueryException(
+          "A resource request resulted in multiple metacards being returned.");
```

Comment:
✏️ It would be useful to include or log which attribute and value was not unique.